### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'split' using anaconda

### ISSUE THREAD

**Issue by Selithrarion** [Yorko/mlcourse.ai#754]
needed to upgrade threadpoolctl to 3.2
pip install threadpoolctl --upgrade
to fix the error [AttributeError: 'NoneType' object has no attribute 'split'](https://stackoverflow.com/questions/71352354/sklearn-kmeans-is-not-working-as-i-only-get-nonetype-object-has-no-attribute)  in [topic 3](https://mlcourse.ai/book/topic03/topic03_decision_trees_kNN.html) on the line
knn_pred = knn.predict(X_holdout_scaled)
also it was in some previous topic but dont remember where

used anaconda so im suggesting to add a hint or smth to help other people if they face it

**Comment by Selithrarion** [comment]
is it should be like this?  data_**test**[c].fillna(data_**train**[c].mode()[0], inplace=True)
```
# fill missing data

for c in categorical_columns:
    data_train[c].fillna(data_train[c].mode()[0], inplace=True)
    data_test[c].fillna(data_train[c].mode()[0], inplace=True)

for c in numerical_columns:
    data_train[c].fillna(data_train[c].median(), inplace=True)
    data_test[c].fillna(data_train[c].median(), inplace=True)
```

**Comment by Selithrarion** [comment]
https://mlcourse.ai/book/topic06/topic6_feature_engineering_feature_selection.html#statistical-approaches - Statistical approaches section,
`VarianceThreshold(0.7).fit_transform(x_data_generated).shape`
and etc doesnt change the data shape

**Comment by Selithrarion** [comment]
https://mlcourse.ai/book/topic08/assignment08_implement_sgd_regressor.html
Linear regression and Stochastic Gradient Descent section
![image](https://github.com/Yorko/mlcourse.ai/assets/59411497/24b8ad43-0f05-4b56-af5f-379cbbce1349)

**Comment by Selithrarion** [comment]
https://mlcourse.ai/book/topic09/assignment09_time_series.html downloads an html file every time i refresh the page o_o

**Comment by Yorko** [comment]
Better late than never. 

> knn_pred = knn.predict(X_holdout_scaled)

Doesn't reproduce, I'm providing all dependencies with Poetry, those should work. 

> plot_tree(
    dt, feature_names=df_train.columns, filled=True, class_names=["Won't go", "Will go"]
);

Don't see issues here. 

> is it should be like this? data_test[c].fillna(data_train[c].mode()[0], inplace=True)

Yes, filling in data in the test set with statistics calculated with the training set. 

> VarianceThreshold(0.7).fit_transform(x_data_generated).shape
and etc doesnt change the data shape

No, it actually throws away one feature 

<img src="https://habrastorage.org/webt/lb/n4/oo/lbn4oodqn8yjbn5hwkdslmh9ipa.png" />

> Linear regression and Stochastic Gradient Descent section

fixed in #768

> https://mlcourse.ai/book/topic09/assignment09_time_series.html downloads an html file every time i refresh the page o_o

no longer see the issue (another one with Prophet will be fixed later #767)
